web/src/app/modules/shared/components/presentation/overflow-selectors/overflow-selectors.component.html

```diff
@@ -1,12 +1,13 @@
 <div class="selectors-wrapper" *ngIf="selectors">
   <div class="selectors-container">
     <ng-container
-      *ngFor="let selector of showSelectors; trackBy: trackByIdentity"
+      *ngFor="let selector of showSelectors; trackBy: trackByIdentity" [ngSwitch]="selector.metadata.type"
     >
-      <app-octant-tooltip tooltipText="{{ selector.config.key }}:{{ selector.config.value }}">
-        <span class="label label-orange">
-          {{ selector.config.key }}:{{ selector.config.value }}
-        </span>
+      <app-octant-tooltip *ngSwitchCase="'labelSelector'" tooltipText="{{ selector.config.key }}:{{ selector.config.value }}">
+        <app-view-label-selector [view]="selector"></app-view-label-selector>
+      </app-octant-tooltip>
+      <app-octant-tooltip *ngSwitchCase="'expressionSelector'" tooltipText="{{ selector.config.key }} {{ selector.config.operator }} {{ selector.config.values?.join(',') }}">
```

Comment:
I'd prefer to see this join as a pipe

Reply:
Sorry if I was unclear, this was meant to be https://angular.io/guide/pipes#using-a-pipe-in-a-template

The reason why we do this is because a pure pipe ensures Angular only re-renders if inputs change rather than relying on the component's change detection strategy.

Reply:
My bad, I didn't know about Angular pipes. Thanks for teaching me something new! I've added a `JoinPipe`, and reverted the delimiter to be a comma